docs/guide/README.md

```diff
@@ -0,0 +1,120 @@
+```bash
+...
+
+# get original dataset
+train_dataset = lm_datasets["train"]
+eval_dataset = lm_datasets["validation"]
+
+# convert values in the dict to torch.tensor
+train_dataset = [
+    {k: torch.tensor(v, dtype=torch.long) for k, v in d.items()}
+    for d in train_dataset
+]
+eval_dataset = [
+    {k: torch.tensor(v, dtype=torch.long) for k, v in d.items()}
+    for d in eval_dataset
+]
+
+# dataloaders
+train_dataloader = DataLoader(
+    train_dataset,
+    ...
+)
+eval_dataloader = DataLoader(
+    eval_dataset,
+    ...
+)
+
+...
+```
+
+## Definition of loss/target function
+The loss/target function to be passed in Attributor requires great care in the dimensionality of data passed in. The reason, in short, is that we sometimes placed a dummy leading dimension for different Attributors to work properly.
+
+### Influence Function
+In Influence Function, we have placed the dummy leading dimension (batch) for loss functions such as `CrossEntropyLoss` to work. An example loss function looks like this:
+```bash
+def f(params, data_target_pair):
+    image, label = data_target_pair
+    loss = nn.CrossEntropyLoss()
+    yhat = torch.func.functional_call(model, params, image)
+    return loss(yhat, label.long())
+```
+On the other hand, if there are any loss functions that don't need a dummy dimension, you may need to manually delete the dummy dimension we have defined in the loss function. An example looks like this:
+```bash
+def f(params, batch):
+    input_ids = batch["input_ids"].squeeze(0).long().cuda()
+    attention_mask = batch["attention_mask"].squeeze(0).long().cuda()
+    labels = batch["labels"].squeeze(0).long().cuda()
+
+    outputs = torch.func.functional_call(
+        model,
+        params,
+        input_ids,
+        kwargs={"attention_mask": attention_mask, "labels": labels},
+    )
+    logp = -outputs.loss
+    return logp - torch.log(1 - torch.exp(logp))
+```
+
+### TRAK or TracIn
+It is worth noting that TRAK and TracIn work the opposite way, where we don't place the dummy dimension. Thus, an example for `CrossEntropyLoss` in TRAK looks like this:
+```bash
+def f(params, data_target_pair):
+    image, label = data_target_pair
+    image_t = image.unsqueeze(0)
+    label_t = label.unsqueeze(0)
+    loss = nn.CrossEntropyLoss()
+    yhat = torch.func.functional_call(model, params, image_t)
+    logp = -loss(yhat, label_t)
+    return logp - torch.log(1 - torch.exp(logp))
+```
+If there are any loss functions that don't need a dummy dimension, you can reference to this example:
+```bash
+def f(params, batch):
+    input_ids, attention_mask, labels = batch
+
+    input_ids = input_ids.cuda()
+    attention_mask = attention_mask.cuda()
+    labels = labels.cuda()
+
+    outputs = torch.func.functional_call(
+        model,
+        params,
+        input_ids,
+        kwargs={"attention_mask": attention_mask, "labels": labels},
+    )
+    logp = -outputs.loss
+    return logp - torch.log(1 - torch.exp(logp))
+```
```

Comment:
Slightly different requirements should be applied to TRAK (Multi-class Margin) and TracIN (training loss and any target function).